tests/math/test_sparse_matrix.py

```diff
@@ -0,0 +1,221 @@
+        s3 = s1 - s2
+        assert s3.data == {k: converter(v) for k, v in res.items()}
+
+    @pytest.mark.parametrize("converter", converters)
+    @pytest.mark.parametrize("matrix1, scalar, res", MULTIPLICATION)
+    def test_multiplication(self, converter, matrix1, scalar, res):
+        """Test scalar multiplication in all frameworks."""
+        tensor1 = converter(matrix1)
+        s1 = SparseMatrix(tensor1)
+        s3 = s1 * scalar
+        s4 = scalar * s1
+        assert s3.data == {k: converter(v) for k, v in res.items()}
+        assert s4 == s3
+
+    @pytest.mark.parametrize("converter", converters)
+    def test_multiplication_zero(self, converter):
+        """Test scalar multiplication with zero."""
+        tensor1 = converter([[1.0, 2.0], [1.0, 0.0]])
+        s1 = SparseMatrix(tensor1)
+        s3 = s1 * 0.0
+        s4 = 0.0 * s1
+        assert s3.data == {}
+        assert s3.shape == (2, 2)
+        assert s4 == s3
+
+    @pytest.mark.parametrize("converter", converters)
+    @pytest.mark.parametrize("arg1, arg2, res", EQUALITY)
+    def test_equality(self, converter, arg1, arg2, res):
+        """Test equality in all frameworks."""
+        tensor1 = converter(arg1)
+        tensor2 = converter(arg2)
+        s1 = SparseMatrix(tensor1)
+        s2 = SparseMatrix(tensor2)
+        eq = s1 == s2
+        assert eq == res
+
+    def test_equality_tuple(self):
+        """Test equality when initializing from tuple."""
+        s1 = SparseMatrix((2, 3))
+        s2 = SparseMatrix((2, 3))
+        assert s1 == s2
+
+        s1 = SparseMatrix((3, 2))
+        s2 = SparseMatrix((2, 3))
+        assert not (s1 == s2)
+
+    def test_equality_different_opject(self):
+        """Test that error raised when trying to compare matrices of different types"""
+        tensor1 = np.array([[1.0, 0.0], [0.0, 0.0]])
+        tensor2 = [[1.0, 0.0], [0.0, 0.0]]
+        s1 = SparseMatrix(tensor1)
+        assert not (s1 == tensor2)
+
+    @pytest.mark.parametrize("converter", converters)
+    @pytest.mark.parametrize("matrix1, matrix2, res, shape", KRON)
+    def test_kron(self, converter, matrix1, matrix2, res, shape):
+        """Test kronecker product in all frameworks."""
+        tensor1 = converter(matrix1)
+        tensor2 = converter(matrix2)
+        s1 = SparseMatrix(tensor1)
+        s2 = SparseMatrix(tensor2)
+        s3 = s1.kron(s2)
+        assert s3.data == {k: converter(v) for k, v in res.items()}
+        assert s3.shape == shape
+
+    def test_kron_error(self):
+        """Test that error raised when trying to compute the kronecker product of matrices of different types"""
+        tensor1 = np.array([[1.0, 0.0], [0.0, 0.0]])
+        tensor2 = np.array([[1.0, 0.0, 0.0], [0.0, 0.0, 0.0]])
+        s1 = SparseMatrix(tensor1)
+        with pytest.raises(
+            ValueError, match="an only compute the kronecker product with another SparseMatrix"
+        ):
+            s1.kron(tensor2)
```

Comment:
only thing missing here are gradient tests (just to ensure that the processing within the sparse matrix class does not break gradient computation).

I know it's tested later in the VQE tests, but might be worth having here. That way, if the processing in the SparseMatrix does cause an issue later, it is easier to track down where the failure is coming from

Reply:
Ah good catch. Oh my!

Reply:
I don't need this class to be differentiable for the current code, but I get what you say. Yo, this is another day of work :cry: